src/main/java/frc/robot/subsystems/Hatch.java

```diff
@@ -26,66 +26,83 @@
 
   /*
    * TODO: confirm with the design team about the type and number of solenoids
    * being used
+   * TODO: Verify the CAN id for the PCM and what voltage level is being used
    */
 
   public Hatch() {
-    this.m_launcher = new Solenoid(RobotMap.m_pcm24v, RobotMap.m_launcherExtend);
+    this.m_launcher = new Solenoid(RobotMap.m_pcm12v, RobotMap.m_launcherExtend);
     this.m_beak = new DoubleSolenoid(RobotMap.m_pcm12v, RobotMap.m_beakExtend, RobotMap.m_beakRetract);
   }
 
-  // simple solenoid operation: push out and turn off immediately
-  public void pneumaticsAutoRetract(DoubleSolenoid solenoid) {
-    solenoid.set(DoubleSolenoid.Value.kForward);
-    try {
-      TimeUnit.SECONDS.sleep(1);
-    } catch (InterruptedException e) {
-      e.printStackTrace();
-    }
-    solenoid.set(DoubleSolenoid.Value.kReverse);
+  @Override
+  public void initDefaultCommand() {
+    // Beak need to be in retracted position first before game.
+    setDefaultCommand(new ExtendBeak());
   }
 
-  public void pneumaticsAutoRetract(Solenoid solenoid){
+  /* Single Solenoid Helper functions */
+  public void pneumaticsOn(Solenoid solenoid) {
     solenoid.set(true);
-    try {
-      TimeUnit.SECONDS.sleep(1);
-    } catch (InterruptedException e) {
-      e.printStackTrace();
-    }
+  }
+
+  public void pneumaticsOff(Solenoid solenoid) {
     solenoid.set(false);
   }
 
-  public void pneumaticsExtend(DoubleSolenoid solenoid){
+  public void toggleSolenoid(Solenoid solenoid) {
+    if (solenoid.get() == true) {
+        pneumaticsOff(solenoid);
+    } else {
+        pneumaticsOn(solenoid);
+    }
+  }
+
+  /* Double Solenoid Helper functions */
+  public void pneumaticsExtend(DoubleSolenoid solenoid) {
     solenoid.set(DoubleSolenoid.Value.kForward);
   }
 
   public void pneumaticsRetract(DoubleSolenoid solenoid) {
     solenoid.set(DoubleSolenoid.Value.kReverse);
   }
 
+  public void toggleSolenoid(DoubleSolenoid solenoid) {
+    if (solenoid.get().equals(DoubleSolenoid.Value.kForward)) {
+        pneumaticsRetract(solenoid);
+    } else {
+        pneumaticsExtend(solenoid);
+    }
+  }
+
+  /* Beak actions */
   public void toggleBeak() {
-    if (m_beak.get().equals(DoubleSolenoid.Value.kForward))
-      pneumaticsRetract(this.m_beak);
-    else
-      pneumaticsExtend(this.m_beak);
+    toggleSolenoid(this.m_beak);
+  }
+
+  public void extendBeak() {
+    pneumaticsExtend(this.m_beak);
   }
 
+  public void retractBeak() {
+    pneumaticsRetract(this.m_beak);
+  }
+
+  /* Launcher actions */
   public void launchHatch() {
-    pneumaticsAutoRetract(this.m_launcher);
+    pneumaticsOn(this.m_launcher);
   }
 
-  @Override
-  public void initDefaultCommand() {
-    // Set the default command for a subsystem here.
+  public void disableLauncher() {
+    pneumaticsOff(this.m_launcher);
+  }
 
-    //Command toggles beak position. Beak need to be in retracted position first before game.
-    setDefaultCommand(new BeakSet()); 
```

Comment:
NOTE: This is how the original continuous clicking bug started - BeakSet() called "toggleBeak()" when it really just need to retract it at the start the match only 1 time.